src/AzureAuth/ExceptionListToStringConverter.cs

```diff
@@ -0,0 +1,133 @@
+        /// <returns>List of correlation IDs.</returns>
+        public static List<string> ExtractCorrelationIDsFromException(IList<Exception> exceptions)
+        {
+            var correlationIDs = new List<string>();
+
+            if (exceptions == null)
+            {
+                return null;
+            }
+
+            foreach (Exception exception in exceptions)
+            {
+                if (exception.GetType() == typeof(MsalServiceException))
+                {
+                    var msalServiceException = (MsalServiceException)exception;
+                    if (msalServiceException.CorrelationId != null)
+                    {
+                        correlationIDs.Add(msalServiceException.CorrelationId);
+                    }
+                }
+                else if (exception.GetType() == typeof(MsalUiRequiredException))
+                {
+                    var msalUiRequiredException = (MsalUiRequiredException)exception;
+                    if (msalUiRequiredException.CorrelationId != null)
+                    {
+                        correlationIDs.Add(msalUiRequiredException.CorrelationId);
+                    }
+                }
+            }
+
+            return correlationIDs;
+        }
+
+        /// <summary>
+        /// Executes the convertion of exceptions to a JSON string.
+        /// </summary>
+        /// <param name="exceptions">List of exceptions.</param>
+        /// <returns>JSON format of exception list.</returns>
+        public static string Execute(IEnumerable<Exception> exceptions)
+        {
+            if (exceptions == null || exceptions.Count() == 0)
+            {
+                return null;
+            }
+
+            List<SerializableException> customExceptions = new List<SerializableException>();
+            foreach (Exception exception in exceptions)
+            {
+                var customException = new SerializableException(exception);
+                customExceptions.Add(customException);
+            }
+
+            return JsonSerializer.Serialize(customExceptions);
+        }
+    }
+
+    /// <summary>
+    /// CustomException class with only select properties of Exception class.
+    /// </summary>
+    internal class SerializableException
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SerializableException"/> class.
+        /// </summary>
+        /// <param name="exception">Exception.</param>
+        public SerializableException(Exception exception)
+        {
+            if (exception != null)
+            {
+                if (exception.Message != null)
+                {
+                    this.Message = exception.Message.Replace("\r", string.Empty);
+                }
+
+                if (exception.InnerException != null)
+                {
+                    this.InnerException = new SerializableException(exception.InnerException);

```

Comment:
If we can, let's document that this is a recursive call. It wasn't immediately obvious to me at a glance that that's how this was implemented.